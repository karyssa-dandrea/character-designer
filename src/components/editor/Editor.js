import React from 'react';

export default function Editor({
  head,
  setHead,
  middle,
  setMiddle,
  bottom,
  setBottom,
  phrase,
  setPhrase,
}) {
  return (
    <div className="editor">
      <div className="form-control">
        <label htmlFor="head">Head</label>
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="bird-head">Bird Head</option>
          <option value="dog-head">Dog Head</option>
          <option value="duck-head">Duck Head</option>
          <option value="horse-head">Horse Head</option>
        </select>
      </div>
      <div className="form-control">
        <select value={middle} onChange={(e) => setMiddle(e.target.value)} />
        <label htmlFor="middle">Middle</label>
      </div>
      <div className="form-control">
        <select value={bottom} onChange={(e) => setBottom(e.target.value)} />
        <label htmlFor="bottom">Bottom</label>
      </div>
      <div className="form-control">
        <input
          name="phrase"
          type="text"
          value={phrase}
          onChange={(e) => setPhrase(e.target.value)}
        />
        <label htmlFor="phrase">Add A Catch Phrase</label>
      </div>
    </div>
  );
}
