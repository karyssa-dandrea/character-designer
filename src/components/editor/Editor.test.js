import { render, screens } from '@testing-library/react';
import Editor from './Editor';

test('renders the editor panel', () => {
  const container = render(<Editor head="" middle="" bottom="" phrase="true" />);
  expect(container).toMatchSnapshot();
});
