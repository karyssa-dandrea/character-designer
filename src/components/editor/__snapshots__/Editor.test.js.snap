// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the editor panel 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="editor"
      >
        <div
          class="form-control"
        >
          <label
            for="head"
          >
            Head
          </label>
          <select>
            <option
              value="bird-head"
            >
              Bird Head
            </option>
            <option
              value="dog-head"
            >
              Dog Head
            </option>
            <option
              value="duck-head"
            >
              Duck Head
            </option>
            <option
              value="horse-head"
            >
              Horse Head
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <label
            for="middle"
          >
            Middle
          </label>
          <select>
            <option
              value="blue"
            >
              Blue
            </option>
            <option
              value="pink"
            >
              Pink
            </option>
            <option
              value="red"
            >
              Red
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <label
            for="bottom"
          >
            Bottom
          </label>
          <select>
            <option
              value="leg-pants"
            >
              Single Leg
            </option>
            <option
              value="white-pants"
            >
              White Pants
            </option>
            <option
              value="darkblue-pants"
            >
              Blue Jeans
            </option>
            <option
              value="dog-pants"
            >
              Dog Pants
            </option>
          </select>
        </div>
        <div
          class="form-control"
        >
          <label
            for="phrase"
          >
            Add A Catch Phrase
          </label>
          <input
            name="phrase"
            type="text"
            value="true"
          />
          <button>
            Add!
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="editor"
    >
      <div
        class="form-control"
      >
        <label
          for="head"
        >
          Head
        </label>
        <select>
          <option
            value="bird-head"
          >
            Bird Head
          </option>
          <option
            value="dog-head"
          >
            Dog Head
          </option>
          <option
            value="duck-head"
          >
            Duck Head
          </option>
          <option
            value="horse-head"
          >
            Horse Head
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <label
          for="middle"
        >
          Middle
        </label>
        <select>
          <option
            value="blue"
          >
            Blue
          </option>
          <option
            value="pink"
          >
            Pink
          </option>
          <option
            value="red"
          >
            Red
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <label
          for="bottom"
        >
          Bottom
        </label>
        <select>
          <option
            value="leg-pants"
          >
            Single Leg
          </option>
          <option
            value="white-pants"
          >
            White Pants
          </option>
          <option
            value="darkblue-pants"
          >
            Blue Jeans
          </option>
          <option
            value="dog-pants"
          >
            Dog Pants
          </option>
        </select>
      </div>
      <div
        class="form-control"
      >
        <label
          for="phrase"
        >
          Add A Catch Phrase
        </label>
        <input
          name="phrase"
          type="text"
          value="true"
        />
        <button>
          Add!
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
